opensquirrel/exporter/quantify_scheduler_exporter.py

```diff
@@ -6,18 +6,18 @@
 from opensquirrel.circuit import Circuit
 from opensquirrel.common import ATOL
 from opensquirrel.default_gates import X, Z
-from opensquirrel.ir import IR, BlochSphereRotation, ControlledGate, IRVisitor, MatrixGate, Measure, Qubit
+from opensquirrel.ir import BlochSphereRotation, ControlledGate, IRVisitor, MatrixGate, Measure, Qubit
 
 try:
     import quantify_scheduler
     import quantify_scheduler.operations.gate_library as quantify_scheduler_gates
-except Exception as e:
+except ModuleNotFoundError:
     pass
 
-_unsupported_gates_exception = Exception(
-    "Cannot export circuit: it contains unsupported gates. "
-    "Decompose them to the Quantify-scheduler gate set first (rxy, rz, cnot, cz)"
-)
+
+class UnsupportedGatesError(Exception):
+    """Should be raised when the set of gates is not supported by the exporter."""
+
```

Comment:
```suggestion

class UnsupportedGateError(Exception):
    """Should be raised when the gate is not supported by the exporter."""

```

Because every time we raise it we will be complaining about a specific gate, right?

Reply:
You're right, its fixed now.